spring-cloud-starter-stream-sink-router/src/main/java/org/springframework/cloud/stream/app/router/sink/RouterSinkConfiguration.java

```diff
@@ -54,7 +59,20 @@ public MessageRouter router(BinderAwareChannelResolver channelResolver,
 			router = new MethodInvokingRouter(scriptProcessor(scriptVariableGenerator, properties));
 		}
 		else {
-			router = new ExpressionEvaluatingRouter(properties.getExpression());
+			router = new ExpressionEvaluatingRouter(properties.getExpression()) {
+				@Override
+				protected void handleMessageInternal(Message<?> message) {
+					if (message.getPayload() instanceof byte[]){
```

Comment:
Since we wanted to port the same logic what we did in the `log-sink` [see [here](https://github.com/spring-cloud-stream-app-starters/log/blob/master/spring-cloud-starter-stream-sink-log/src/main/java/org/springframework/cloud/stream/app/log/sink/LogSinkConfiguration.java#L53)], let's move ahead with this change.